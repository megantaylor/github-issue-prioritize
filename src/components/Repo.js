import * as React from 'react';
import styled from 'styled-components';

const RepoContainer = styled.div`
  &.active {
    background: yellow;
  }
  cursor: default;
`;

const Title = styled.h3`
  @media (max-width: 600px) {
    word-break: break-word;
  }
`;

export default function Repo(props) {
  return (
    <RepoContainer
      onClick={() =>
        props.handleRepoClick(props.repo.issues_url.slice(0, -9), props.repo.id)
      }
      className={props.className}
    >
      <Title>{props.repo.full_name}</Title>
    </RepoContainer>
  );
}
