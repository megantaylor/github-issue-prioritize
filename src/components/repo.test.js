import React from 'react';
import Repo from './Repo';
import renderer from 'react-test-renderer';

it('renders inactive repo correctly', () => {
  const tree = renderer
    .create(<Repo repo={{ full_name: 'repo title' }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders active repo correctly', () => {
  const tree = renderer
    .create(<Repo repo={{ full_name: 'repo title' }} className='active' />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
