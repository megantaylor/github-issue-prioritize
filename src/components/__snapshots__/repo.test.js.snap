// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders active repo correctly 1`] = `
<div
  className="active sc-bdVaJa epmeOv"
  onClick={[Function]}
>
  <h3
    className="sc-bwzfXH fHVcXE"
  >
    repo title
  </h3>
</div>
`;

exports[`renders inactive repo correctly 1`] = `
<div
  className="sc-bdVaJa epmeOv"
  onClick={[Function]}
>
  <h3
    className="sc-bwzfXH fHVcXE"
  >
    repo title
  </h3>
</div>
`;
